contracts/kernel/Kernel.sol

```diff
@@ -54,6 +56,27 @@ contract Kernel is IKernel, KernelStorage, KernelAppIds, KernelNamespaceConstant
         recoveryVaultAppId = KERNEL_DEFAULT_VAULT_APP_ID;
     }
 
+    /**
+    * @dev Initialize can only be called once. It saves the block number in which it was initialized.
+    * @notice Initialize this kernel instance, its ACL setting `_permissionsCreator` as the entity that can create other permissions, and a KillSwitch instance setting `_issuesRegistry`
+    * @param _baseAcl Address of base ACL app
+    * @param _permissionsCreator Entity that will be given permission over createPermission
+    * @param _baseKillSwitch Address of base KillSwitch app
+    * @param _issuesRegistry Issues registry that will act as the default source of truth to provide info about applications issues
+    */
+    function initializeWithKillSwitch(IACL _baseAcl, address _permissionsCreator, IKillSwitch _baseKillSwitch, IIssuesRegistry _issuesRegistry)
+        public onlyInit
+    {
+        // Set and create ACL app
+        initialize(_baseAcl, _permissionsCreator);
+
+        // Set and create KillSwitch app
+        _setApp(KERNEL_APP_BASES_NAMESPACE, KERNEL_DEFAULT_KILL_SWITCH_APP_ID, _baseKillSwitch);
+        IKillSwitch killSwitch = IKillSwitch(newAppProxy(this, KERNEL_DEFAULT_KILL_SWITCH_APP_ID));
+        killSwitch.initialize(_issuesRegistry);
+        _setApp(KERNEL_APP_ADDR_NAMESPACE, KERNEL_DEFAULT_KILL_SWITCH_APP_ID, killSwitch);
+    }
```

Comment:
To avoid coupling the Kernel too much with the KillSwitch, I think this bit of logic could be moved higher up the stack to the `DAOFactory`.

I think this would also make it more explicit that your Kernel can function without a KillSwitch, and that you can install and remove the KillSwitch as wanted.